GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/GarageController.cs

```diff
@@ -0,0 +1,91 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+
+// controller of the garage page
+public class GarageController : MonoBehaviour
+{
+    public GameObject playCarGameObject;
+    public Text coinsCountText;
+
+    private GameManager _gameManager;
+    private GameData _gameData;
+    private PlayerController _playerController;
+
+
+    private void Awake()
+    {
+        _gameManager = FindObjectOfType<GameManager>();
+        _gameData = _gameManager.GetGameData();
+        _playerController = playCarGameObject.GetComponent<PlayerController>();
+    }
+
+
+    // fresh the page when on eable
+    private void OnEnable()
+    {
+        RefreshPage();
+    }
+
+    private void RefreshPage()
+    {
+        // check if player already owns the car
+        CheckCarOwnership();
+        CheckUsingStatus();
+        SetCoins();
+    }
+
+    private void CheckCarOwnership()
+    {
+        foreach (var car in CarList.List)
+        {
+            var isCarOwned = _gameData.CheckOwnership(car.carName);
+            // set the car item active if the player owns the car.
+            car.garageItemGameObj.SetActive(isCarOwned);
+        }
+    }
+
+    private void CheckUsingStatus()
+    {
+        foreach (var carObj in CarList.List)
+        {
+            carObj.garageItemGameObj.transform.Find("statusText").gameObject.SetActive(false);
+        }
+
+        _gameData.GetCarInUseObj().garageItemGameObj.transform.Find("statusText").gameObject.SetActive(true);
+    }
+
+
+    public void OnItemSedanClicked()
+    {
+        SwitchCarInUse(CarList.CarSedan);
+    }
+
+    public void OnItemTruckClicked()
+    {
+        SwitchCarInUse(CarList.CarTruck);
+    }
+
+    public void OnItemJeepClicked()
+    {
+        SwitchCarInUse(CarList.CarJeep);
+    }
+
+    public void OnItemKartClicked()
+    {
+        SwitchCarInUse(CarList.CarKart);
+    }
+
+    private void SwitchCarInUse(Car targetCar)
+    {
+        _gameData.carInUseName = targetCar.carName;
+        _gameData.SetCarInUseObj();
```

Comment:
Use a public method, instead of modifying the variable directly
_gameData.ChangeCar(targetCar)

Reply:
Or SelectCar